pangeo_forge_recipes/writers.py

```diff
@@ -10,12 +10,14 @@
 def _region_for(var: xr.Variable, index: Index) -> Tuple[slice, ...]:
     region_slice = []
     for dim, dimsize in var.sizes.items():
-        concat_dim_val = index.find_concat_dim(dim)
-        if concat_dim_val:
+        concat_dimension = index.find_concat_dim(dim)
+        if concat_dimension:
             # we are concatenating over this dimension
-            assert concat_dim_val.start is not None
-            assert concat_dim_val.stop == concat_dim_val.start + dimsize
-            region_slice.append(slice(concat_dim_val.start, concat_dim_val.stop))
+            position = index[concat_dimension]
+            assert position.indexed
```

Comment:
Would be useful to have an assert message, maybe with some message to help end-users debug.